src/phylum/ci/ci_base.py

```diff
@@ -544,6 +551,67 @@ def analyze(self) -> None:
+            return []
+
+        with tempfile.TemporaryDirectory(prefix="phylum_") as temp_dir:
+            cmd = ["git", "worktree", "add", "--detach", temp_dir, self.common_ancestor_commit]
+            LOG.debug("Adding a git worktree for the base iteration in a temporary directory ...")
+            LOG.debug("Using command: %s", shlex.join(cmd))
+            try:
+                subprocess.run(cmd, check=True, capture_output=True, text=True)  # noqa: S603
+            except subprocess.CalledProcessError as err:
+                pprint_subprocess_error(err)
+                LOG.error("Due to error, assuming no base dependencies. Please report this as a bug.")
+                return []
+
+            temp_dir_path = Path(temp_dir).resolve()
+            base_packages = set()
+            for depfile in self.depfiles:
+                prev_depfile_path = temp_dir_path / depfile.path.relative_to(Path.cwd())
+                cmd = [str(self.cli_path), "parse", "--lockfile-type", depfile.type, str(prev_depfile_path)]
+                LOG.info(
+                    "Parsing [code]%r[/] as previous [code]%s[/] [b]%s[/] ...",
+                    depfile,
+                    depfile.type,
+                    depfile.depfile_type.value,
+                    extra=MARKUP,
+                )
+                if depfile.is_manifest:
+                    LOG.info("  This may take a while")
+                LOG.debug("Using parse command: %s", shlex.join(cmd))
+                try:
+                    parse_result = subprocess.run(
+                        cmd,  # noqa: S603
+                        cwd=temp_dir_path,
+                        check=True,
+                        capture_output=True,
+                        text=True,
+                    ).stdout.strip()
+                except subprocess.CalledProcessError as err:
+                    pprint_subprocess_error(err)
+                    msg = f"""\
+                        Due to error, assuming no previous [b]{depfile.depfile_type.value}[/] packages.
+                          Consider supplying lockfile type explicitly in `.phylum_project` file.
+                          For more info, see: https://docs.phylum.io/docs/lockfile_generation
+                          Please report this as a bug if you believe [code]{depfile!r}[/]
+                          is a valid [code]{depfile.type}[/] [b]{depfile.depfile_type.value}[/] at revision
+                          [code]{self.common_ancestor_commit}[/]."""
+                    LOG.warning(textwrap.dedent(msg), extra=MARKUP)
+                    remove_git_worktree(temp_dir_path)
+                    return []
+
+                parsed_pkgs = json.loads(parse_result)
+                prev_depfile_packages = [PackageDescriptor(**pkg) for pkg in parsed_pkgs]
+                base_packages.update(prev_depfile_packages)
+
+            remove_git_worktree(temp_dir_path)
```

Comment:
It might be nice to abstract the worktree create/delete into a context manager. You could even create the tempdir in the context manager. So this code would become:

```py
base_packages = set()
with git_worktree(self.ancestor_commit) as tmp_dir:
    for depfile in self.depfiles:
        pass # Analyze the file
return sorted(base_packages)
```

And you'd define `git_worktree` something like so (error handling and logging omitted for brevity)

```py
@contextlib.contextmanager
def git_worktree(commit: str):
    with tempfile.TemporaryDirectory(prefix="phylum_") as temp_dir:
        cmd = ["git", "worktree", "add", "--detach", temp_dir, commit]
        subprocess.run(cmd, check=True, capture_output=True, text=True)
        try:
            yield temp_dir
        finally:
            remove_git_worktree(temp_dir)
```

Reply:
Done...plus a little bit more.